cmd/parca-agent/main.go

```diff
@@ -114,11 +114,19 @@ type Profiler interface {
 }
 
 func main() {
+	hostname, hostnameErr := os.Hostname()
+
 	flags := flags{}
-	kong.Parse(&flags)
+	kong.Parse(&flags, kong.Vars{
+		"hostname": hostname,
+	})
 
 	logger := logger.NewLogger(flags.LogLevel, logger.LogFormatLogfmt, "parca-agent")
 
+	if flags.Node == "" && hostnameErr != nil {
+		level.Error(logger).Log("msg", "failed to get host name. Please set it with the --node flag", "err", hostnameErr)
```

Comment:
```suggestion
		level.Error(logger).Log("msg", "failed to get host name. Please set it with the --node flag", "err", hostnameErr)
		os.Exit(1)
```

We should fail early in this case. Or we can move this check into the `run` and just return an error.

Reply:
Good catch! I intended to fail here